src/App.tsx

```diff
@@ -1,60 +1,133 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(users => users.id === todo.userId) || null,
+}));
+
+const usersFromServerFilter = usersFromServer
+  .find(users => todosFromServer
+    .find(todo => users.id === todo.userId)) || null;
 
 export const App = () => {
+  const [name, setName] = useState('');
+  const [userSelect, setUserSelect] = useState(usersFromServerFilter);
+  const [todos, setTodos] = useState<Todo[]>([...preparedTodos]);
+
+  const [formKey, setFormKey] = useState(10);
+  const [isErrorName, setIsErrorName] = useState(false);
+  const [isErrorUser, setIsErrorUser] = useState(false);
+  const [currentUserId, setCurrentUserId] = useState(0);
+
+  const addTodo = (title: string, user: User | null) => {
+    const newTodo = {
+      id: preparedTodos.length + 1,
+      userId: 1,
+      title,
+      user,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsErrorName(() => (name === ''));
+
+    setIsErrorUser(() => (currentUserId === 0));
+
+    if (name !== '' && currentUserId !== 0) {
+      addTodo(name, userSelect);
+      setName('');
+      setFormKey(formKey + 1);
+    }
+  };
+
+  const handle = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    event.preventDefault();
+    const { value } = event.target;
+    const needUser = usersFromServer.find(
+      item => item.name === value,
+    );
+
+    setUserSelect(needUser || null);
+    setCurrentUserId(1);
+  };
+
+  const errorTitle = <span className="error">Please enter a title</span>;
+  const errorUser = <span className="error">Please choose a user</span>;
```

Comment:
```suggestion
  const errorTitle = Please enter a title;
  const errorUser = Please choose a user;
```
If you need class then add this span to jsx and than show or not 

```suggestion
     ....JSX
 
     {isErrorTitle && <span className="error">{errorTitle}</span>}
     {isErrorUser && <span className="error">{errorUser}</span>}
```